src/number/complete.rs

```diff
@@ -1577,7 +1627,44 @@ where
 /// assert_eq!(parser("123K-01"), Ok(("K-01", 123.0)));
 /// assert_eq!(parser("abc"), Err(Err::Error(("abc", ErrorKind::Float))));
 /// ```
-pub fn double<T, E: ParseError<T>>(input: T) -> IResult<T, f64, E>
+pub fn float<T, E: ParseError<T>>(input: T) -> IResult<T, f32, E>
+where
+  T: Slice<RangeFrom<usize>> + Slice<RangeTo<usize>> + Slice<Range<usize>>,
+  T: Clone + Offset,
+  T: InputIter + InputLength + InputTake,
+  <T as InputIter>::Item: AsChar + Copy,
+  <T as InputIter>::IterElem: Clone,
+  T: InputTakeAtPosition,
+  <T as InputTakeAtPosition>::Item: AsChar,
+  T: AsBytes,
+  T: for<'a> Compare<&'a [u8]>,
+{
+  if let Ok((i, float)) = float_finite::<T, E>(input.clone()) {
+    Ok((i, float))
+  } else {
+    float_nonfinite::<T, E>(input)
+  }
```

Comment:
use `or_else` ? or `alt` parser

Reply:
Sure, that works out. If we're moving the sign parsing out though, that might not be possible anymore.

Reply:
Done, I use the `Result.or` here, since it works exactly with what we want to do. For the streaming parser, I still use a match statement.

https://github.com/Alexhuszagh/nom/blob/805e5be9d7e16a6fc0d86e3e6a16fd042d97fd3f/src/number/complete.rs#L1623-L1625

Reply:
`or` will always call `float_nonfinite`